### PR TITLE
Set codecs at constructor

### PR DESCRIPTION
**Pull request by rochoa**
Configuring the codecs should happen at the constructor because we already have the metadata.

However, this doesn't work because the codecs hold a reference to the metadata. Cloning the source will fail with a JSON circular reference in https://github.com/CartoDB/carto-vl/blob/a2fdec46c9707a3b7b3ab0cb584a256d37ebe9cd/src/sources/MVT.js#L75-L77

We already had [a branch that was removing those references](https://github.com/CartoDB/carto-vl/tree/codecs-metadata-avoid-references). The initial proposition was more about performance, but now we have a better reason to merge that branch: being able to clone the sources without that circular structure issue and configuring the codecs at the right place.

What do you think?

### CODE REVIEW

**Comment by rochoa** [comment]
BTW, this will fix the issue described at https://github.com/CartoDB/carto-vl/issues/1216.

File: src/client/windshaft.js [hunk 1]
```diff
@@ -401,9 +401,7 @@ export default class Windshaft {
 
         const idProperty = 'cartodb_id';
 
-        const metadata = new Metadata({ properties, featureCount, sample: stats.sample, geomType, isAggregated: aggregation.mvt, idProperty });
-        metadata.setCodecs();
-        return metadata;
+        return new Metadata({ properties, featureCount, sample: stats.sample, geomType, isAggregated: aggregation.mvt, idProperty });
```
Comment (VictorVelarde) [inline src/client/windshaft.js]: Should this same change be applied to GeoJSON?

Comment (VictorVelarde) [reply src/client/windshaft.js]: If base `Metadata` is not the place anymore to properly `setCodecs`, maybe we could 'defend' from unwanted uses creating there something like:

```js
    setCodecs() {
        throw new CartoRuntimeError(`You must call 'setCodecs' once you have determined the proper subclass`);
    }
```

Does it make sense to you?

Comment (rochoa) [reply src/client/windshaft.js]: I might not be understanding your comment about GeoJSON. We are already doing something similar in the GeoJSONN source, see https://github.com/CartoDB/carto-vl/blob/a2fdec46c9707a3b7b3ab0cb584a256d37ebe9cd/src/sources/GeoJSON.js#L159-L160

Comment (rochoa) [reply src/client/windshaft.js]: Fair point about the deffensive code in the base metadata class. I will add that.

**Comment by rochoa** [comment]
I prefer to discuss #1223 and merge into master instead of mixing the two things in just one PR. If that other PR gets into master, I will update this to check e2e tests.

File: src/client/windshaft.js (hunk 1 above)
Comment (VictorVelarde) [reply src/client/windshaft.js]: Regarding to GeoJSON, I thought both lines would rise this very same 'deffensive error', but I have just seen that Metadata there is already a GeoJSONMetadata, so forget about it.

**Comment by rochoa** [comment]
After merging #1223 into master and updating this PR with master, all tests are green. 🎉

**Pull request merged**